h2o-py/h2o/explanation/_explain.py

```diff
@@ -1326,6 +1330,9 @@ def ice_plot(
             else:
                 plt.plot(tmp[encoded_col], tmp["mean_response"], color=colors[i],
                          label="{}th Percentile".format(i * 10))
+            plt.scatter(frame.as_data_frame()[column][index], model.predict(frame).as_data_frame()['predict'][index],
```

Comment:
It seems rather wasteful to make a prediction on a whole frame in a for loop, especially if only a single value is used always. 
1) Isn't it sufficient to use the PDP values? To me, as an outside spectator, it seems that `partial_plot` with `row_index` specified would be doing the same thing and the prediction for this particular variable value should be matching. Please correct me if that's a wrong assumption
2) If the value from `partial_plot` is wrong, then would it make sense to make just a single prediction on a whole frame, and just reuse it?

Reply:
For a single point (categorical variable) it is matching. For continuous, the point can be outside of its underlying ice line since line is created as an interpolation of a couple of points and that can be imperfect for small nbins. I was discussing this with @tomasfryda and it seemed to us more proper to paint exact predicted value than interpolated value.

prediction should be only 1 and reused, thx 👍

Reply:
@koniecsveta that's not entirely clear to me. Are you saying that predictions are computed on multiple rows and ICE for a specific row is interpolated? Or are you referring to some line smoothening interpolation function?

Reply:
@johnygomez to some line smoothening interpolation function

Reply:
I see, would it make sense to try different interpolation functions then, so that they match points more closely? I'm sure you analyzed that for a longer time, it's just that at first glance it looks rather confusing.

Reply:
when I set nbins to 100, what is the required default value for nbins configurable by the parameter which is to be added as another ICE improvement (https://h2oai.atlassian.net/browse/PUBDEV-8494), I get this much better plot:
![age_both](https://user-images.githubusercontent.com/18303397/151597685-be56e3e8-3a49-4abd-8315-58cf7e5b0ab7.png)

I think I could set nbins=100 already in this PR (currently we are using unconfigurable 20 which is simply insufficient) and add configurability in underlying ticket, WDYT?

Reply:
IMHO setting this to a fixed value won't solve the issue, as it strongly depends on the data. It's however better than before and can be addressed later ofc. When looking at the code, however, are you sure there really is any interpolation? If so, where does it happen?

Reply:
Agree.. I meant ggplot2::geom_line() simple linear interpolation. It is only as descriptive as many points you give it. Currently there is no advanced logic dealing with interpolation in h2o ice_plot code.

Reply:
@koniecsveta but that's only for R, I don't see the interpolation done in matplotlib, so I'd assume there is none then. Still, the points are away from the line.

When looking at the R example, however, it also looks fishy as the ICE line seems to be linear, yet the points are far away. How is that possible? That cannot be explained by the interpolation.
![image](https://user-images.githubusercontent.com/9008237/151781489-83353b13-433c-4905-beb5-a22df0734acc.png)

Reply:
👍 looking into it

Reply:
@johnygomez I've found the issue: df3a79e 

The problem was with indexing. Java indexes from 0 and R from 1, hence if I calculate ICE points on Java backend for let's say `i`, the corresponding row for original value is `i + 1`, not `i` :woman_facepalming:

So currently R output is consistent with py output:
![Rplotage](https://user-images.githubusercontent.com/18303397/152008642-1d3e4f98-b374-453b-bad5-9eaa5cf45b08.png)